### PR TITLE
Include URI summary in search results

### PR DESCRIPTION
**Pull request by martinraison**
@ymatsuda @andrewconner

### CODE REVIEW

**Comment by 2is10** [comment]
why?

**Comment by 2is10** [comment]
(I’m just curious on the motivation, since this will add latency and increase the response size. Thanks.)

**Comment by andrewconner** [comment]
Taking about this over hipchat

**Comment by martinraison** [comment]
I'm going to edit this pull request. I'll add a boolean parameter for the search, that tells the search service to return description/image url for each bookmark

**Comment by andrewconner** [comment]
To summarize Hipchat, I think this is unnecessary. We would only need the external ID if we needed to make further requests to the server about this specific keep, and right now, the only use case we have is for the images + description. Since we can more easily cache, batch, parallelize, and format the request server side, it probably makes more sense for there to be an optional param to include this stuff on the first request. This will add latency to the search speed, but it's what we'll have to deal with either way.

We may need externalId later, but I don't think we need it now (rekeeping is done by URL). Your proposed plan is excellent.

Would love @2is10, @eishay, @ymatsuda feedback.

(This is all regarding needing the image / page description in search results on kifi.com — something we don't have now, and involved a round trip to shoebox)

File: kifi-backend/modules/search/app/com/keepit/common/cache/SearchCacheModule.scala [hunk 1]
```diff
@@ -56,6 +56,11 @@ case class SearchCacheModule(cachePluginModules: CachePluginModule*) extends Cac
 
   @Singleton
   @Provides
+  def normalizedURIExternalIDCache(stats: CacheStatistics, accessLog: AccessLog, outerRepo: FortyTwoCachePlugin) =
+    new NormalizedURIExternalIDCache(stats, accessLog, (outerRepo, 7 days))
```
Comment (ymatsuda) [inline kifi-backend/modules/search/app/com/keepit/common/cache/SearchCacheModule.scala]: no in-memory cache?

**Comment by martinraison** [comment]
I probably wasn't clear enough in my previous comment: I'm actually removing the external id as @andrewconner suggested - but I'm still building on this pull request because most of the changes will actually be fairly similar

File: kifi-backend/modules/search/app/com/keepit/common/cache/SearchCacheModule.scala (hunk 1 above)
Comment (martinraison) [inline kifi-backend/modules/search/app/com/keepit/common/cache/SearchCacheModule.scala]: Sure, I can add it. I wasn't aware we had that. Actually I'm removing this cache and creating a URISummaryCache instead. URISummaries mostly contain a small text description (~200 characters) and an image url. Does that affect the configuration we want to have for the cache?

Comment (andrewconner) [inline kifi-backend/modules/search/app/com/keepit/common/cache/SearchCacheModule.scala]: I say cache the description and image URL very aggressively. In-memory is great, even for longer than others. If this is outdated, it's no problem — and we don't even update it right now. The only concern is memory usage by caching so much in memory. @ymatsuda would probably be able to recommend the best TTL.

Comment (ymatsuda) [inline kifi-backend/modules/search/app/com/keepit/common/cache/SearchCacheModule.scala]: OK. I'd like to have in-memory cache for URISummary.

Comment (ymatsuda) [inline kifi-backend/modules/search/app/com/keepit/common/cache/SearchCacheModule.scala]: TTL for in-memory cache, let's start from 5 minutes and see the cache hit/miss ratio.

@martinraison, please add URISummaryCache to "Cache Overview" admin page.

**Comment by 2is10** [comment]
I’m fine with optionally adding more information to the search result JSON.

Personally, I’d vote for using a slightly different path (a new action) for
searches from the site, rather than adding a new query parameter to the
existing search action to indicate what information is desired.

**Comment by andrewconner** [comment]
@2is10 It already uses a separate action/route for the site. https://search.kifi.com/site/search?f=f&maxHits=30&q=dog

**Comment by 2is10** [comment]
cool

**Comment by ymatsuda** [comment]
lgtm!

**Pull request merged**